### PR TITLE
[OSDOCS-6908] IMDSv2 Documentation wrongly under Node Management

### PR DESCRIPTION
**Pull request by mletalie**
Version(s):

4.13+
Issue:

https://issues.redhat.com/browse/OSDOCS-6908
Link to docs preview:

https://62606--docspreview.netlify.app/openshift-rosa/latest/rosa_cli/rosa-manage-objects-cli.html#rosa-create-cluster-command_rosa-managing-objects-cli
image

https://62606--docspreview.netlify.app/openshift-rosa/latest/rosa_cluster_admin/rosa_nodes/rosa-managing-worker-nodes.html
image

https://62606--docspreview.netlify.app/openshift-rosa/latest/rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-with-customizations.html#rosa-sts-creating-cluster-customizations-cli_rosa-sts-creating-a-cluster-with-customizations
image

https://62606--docspreview.netlify.app/openshift-rosa/latest/rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-with-customizations.html#rosa-sts-creating-cluster-customizations-ocm_rosa-sts-creating-a-cluster-with-customizations
image

https://62606--docspreview.netlify.app/openshift-rosa/latest/rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-quickly.html#rosa-sts-overview-of-the-default-cluster-specifications_rosa-sts-creating-a-cluster-quickly
image

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by mletalie** [comment]
Hi 
[@Xue Li](https://redhat-internal.slack.com/team/UEEQ10T4L)
 and 
[@yuwan](https://redhat-internal.slack.com/team/UEE8X128Y)
. Could I please get a QE review of this PR when you get a chance: https://github.com/openshift/openshift-docs/pull/62606. Thanks, appreciate it!

File: modules/rosa-create-objects.adoc [hunk 1]
```diff
@@ -134,6 +134,9 @@ a|--cluster-name <cluster_name>
 |--dry-run
 |Simulates creating the cluster.
 
+|----ec2-metadata-http-tokens string
```
Comment (sunzhaohua2) [inline modules/rosa-create-objects.adoc:137]: should be `--ec2-metadata-http-tokens`

**Comment by sunzhaohua2** [comment]
For the  `rosa describe cluster ` command, now if we explicitly set imds value, `Ec2 Metadata Http Tokens` will be displayed in the `rosa describe cluster` result, if use default value, `rosa describe cluster` will not show "Ec2 Metadata Http Tokens", in the future if rosa describe will always show “Ec2 Metadata Http Token”, the `rosa describe cluster ` output should be updated.
https://docs.openshift.com/rosa/rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-with-customizations.html#rosa-sts-creating-cluster-customizations-cli_rosa-sts-creating-a-cluster-with-customizations

Need to add `Configure the use of IMDSv2 for ec2 instances optional/required (optional)` in https://docs.openshift.com/rosa/rosa_install_access_delete_clusters/rosa-sts-interactive-mode-reference.html#rosa-sts-interactive-cluster-creation-mode-options_rosa-sts-interactive-mode-reference

**Comment by mletalie** [comment]
For the rosa describe cluster  command, now if we explicitly set imds value, Ec2 Metadata Http Tokens will be displayed in the rosa describe cluster result, if use default value, rosa describe cluster will not show "Ec2 Metadata Http Tokens", in the future if rosa describe will always show “Ec2 Metadata Http Token”, the rosa describe cluster  output should be updated. **(SELF REFERENCE: # 6 Example Output)**
https://docs.openshift.com/rosa/rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-with-customizations.html#rosa-sts-creating-cluster-customizations-cli_rosa-sts-creating-a-cluster-with-customizations

**Comment by mletalie** [comment]
> 

Hello @sunzhaohua2,
   In regards to the comment above, as you can see, I did  set imds value: 
image, but I it is not showing here: 
image
I assume this is still in staging environment, and not production environment? I am using 1.2.23 ROSA CLI in production environment..

**Comment by sunzhaohua2** [comment]
> I assume this is still in staging environment, and not production environment? I am using 1.2.23 ROSA CLI in production environment..

Seems this is not related to environment, we need to either build a rosa cli from code, or wait for a new rosa cli release

**Comment by karanthshashank** [comment]
@bchandra-ocp tagging Bala to review this PR.

**Comment by mletalie** [comment]
Hello @sunzhaohua2, 
In regards to the comment below, please see:  https://62606--docspreview.netlify.app/openshift-rosa/latest/rosa_install_access_delete_clusters/rosa-sts-interactive-mode-reference
image


> Need to add `Configure the use of IMDSv2 for ec2 instances optional/required (optional)` in https://docs.openshift.com/rosa/rosa_install_access_delete_clusters/rosa-sts-interactive-mode-reference.html#rosa-sts-interactive-cluster-creation-mode-options_rosa-sts-interactive-mode-reference

**Comment by mletalie** [comment]
> The original screen-caps [#62606 (comment)](https://github.com/openshift/openshift-docs/pull/62606#issue-1812722533) are fixing the original bug raised i.e., setting the field in day-1.
> 
> rosa describe command that supports this new field requires CLI release. We can either have two different OSDOCs issue respectively for each or hold on this PR for a CLI release that addresses https://issues.redhat.com/browse/OCM-2892 is available.

Hello @bchandra-ocp,
   Based on the comment above, I am going to open another OSDOCS ticket(and assign to myself) that addresses  https://issues.redhat.com/browse/OCM-2892. I can tell you that I just upgraded to the newest CLI version (1.2.24) and the EC2 Metadata HTTP Tokens (ec2_metadata_http_tokens) configuration is still not displayed in the CLI. There is a lot of information already covered in this PR (https://github.com/openshift/openshift-docs/pull/62606) so I want to get it reviewed/merged/closed in this sprint if at all possible.
CC: @sunzhaohua2

**Comment by sunzhaohua2** [comment]
LGTM

File: modules/rosa-imds-ui.adoc [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Module included in the following assemblies:
+//
+// * rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-with-customizations.adoc
+
+:_content-type: PROCEDURE
+[id="rosa-imds-ui_{context}"]
+= Enabling Instance Metadata Service in {cluster-manager}
+
+You can select your Instance Metadata Service (IMDS) type when creating your cluster in {cluster-manager}. You can select to use both IMDSv1 and IMDSv2, or you can specify to only use IMDSv2.
```
Comment (bscott-rh) [inline modules/rosa-imds-ui.adoc:9]:
```suggestion
You can select your Instance Metadata Service (IMDS) type when creating your cluster in {cluster-manager}. You can select both IMDSv1 and IMDSv2, or you can select only IMDSv2.
```
Rewording suggestion

File: modules/rosa-imds-ui.adoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+* You installed and configured the latest AWS (`aws`), ROSA (`rosa`), and OpenShift (`oc`) CLIs on your workstation.
+* You logged in to your Red Hat account by using the `rosa` CLI.
+* You have the permissions to create and manage clusters.
+* You have access to {cluster-manager-url}.
+
+.Procedure
+
+. Log in to the web console.
+. Create a ROSA cluster using your preferences.
+. In the **Create a ROSA Cluster** wizard on the **Cluster settings** -> **Machine pool** page, under the *Instance Metadata Service (IMDS)* section, select whether your machine pools use either both IMDSv1 and IMDSv2 or if you only want to use IMDSv2.
```
Comment (bscott-rh) [inline modules/rosa-imds-ui.adoc:22]:
```suggestion
. In the **Create a ROSA Cluster** wizard on the **Cluster settings** -> **Machine pool** page, under the *Instance Metadata Service (IMDS)* section, select whether your machine pools use both IMDSv1 and IMDSv2, or only IMDSv2.
```
Rewording suggestion

File: modules/rosa-imds-ui.adoc [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+. Select *Next* to save this selection.
+
+.Verification
+
+. After your cluster has been created, on your cluster *Overview* tab, you see the *Instance Metadata Service (IMDS)* field that notes your IMDS version support.
```
Comment (bscott-rh) [inline modules/rosa-imds-ui.adoc:27]:
```suggestion
. After your cluster has been created, see the *Instance Metadata Service (IMDS)* field that notes your IMDS version support on the cluster *Overview* tab.
```
Reworded for clarity

File: modules/rosa-imds.adoc [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+= Instance Metadata Service
+
+There are two types of ways to access instance metadata from a running instance:
+
+* Instance Metadata Service Version 1 (IMDSv1) - a request/response method
+* Instance Metadata Service Version 2 (IMDSv2) - a session-oriented method
+
+IMDSv2 uses session-oriented requests. With session-oriented requests, you create a session token that defines the session duration, which can be a minimum of one second and a maximum of six hours. During the specified duration, you can use the same session token for subsequent requests. After the specified duration expires, you must create a new session token to use for future requests.
+
+When creating your ROSA cluster, you select to use either both IMDSv1 and IMDSv2 or specify that your cluster should only use IMDSv2. The instance metadata service distinguishes between IMDSv1 and IMDSv2 requests based on whether, for any given request, either the PUT or GET headers, which are unique to IMDSv2, are present in that request. If you specify to use IMDSv2 only, IMDSv1 ceases to function for your cluster. All machine pools on your cluster will use whichever IMDS type you select.
```
Comment (bscott-rh) [inline modules/rosa-imds.adoc:16]:
```suggestion
When creating your ROSA cluster, you can configure your cluster to use both IMDSv1 and IMDSv2, or only IMDSv2. The instance metadata service distinguishes between IMDSv1 and IMDSv2 requests based on whether the PUT or GET headers, which are unique to IMDSv2, are present in that request. If you specify to use IMDSv2 only, IMDSv1 ceases to function for your cluster. All machine pools on your cluster will use whichever IMDS type you select.
```
reworded for clarity

File: modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc [hunk 6]
```diff
@@ -223,46 +223,54 @@ $ rosa create cluster --interactive --sts
...
 I: Once the cluster is installed you will need to add an Identity Provider before you can login into the cluster. See 'rosa create idp --help' for more information.
 ...
 ----
-<1> When creating the cluster, the listed `OpenShift version` options include the major, minor, and patch versions, for example `4.8.9`.
-<2> If you have more than one set of account roles in your AWS account for your cluster version, an interactive list of options is provided.
-<3> Optional: By default, the cluster-specific Operator role names are prefixed with the cluster name and random 4-digit hash. You can optionally specify a custom prefix to replace `<cluster_name>-<hash>` in the role names. The prefix is applied when you create the cluster-specific Operator IAM roles. For information about the prefix, see _Defining an Operator IAM role prefix_.
+<1> When creating the cluster, the listed `OpenShift version` options include the major, minor, and patch versions, for example `4.13.4`.
+<2> Optional: By default (optional) all EC2 instances will use both v1 and v2 endpoints of EC2 Instance Metadata Service (IMDS). Enforce to have cluster nodes use IMDSv2 only (required).
```
Comment (bscott-rh) [inline modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc:259]:
```suggestion
<2> Optional: Specify "optional" to configure all EC2 instances to use both v1 and v2 endpoints of EC2 Instance Metadata Service (IMDS). This is the default value. Specify "required" to configure all EC2 instances to use IMDSv2 only.
```
I found this section confusing, so I took a shot at rewording it for clarity. What do you think?

File: modules/rosa-sts-interactive-cluster-creation-mode-options.adoc [hunk 7]
```diff
@@ -25,6 +25,9 @@ The following table describes the interactive cluster creation mode options:
 |`OpenShift version`
 |Select the version of OpenShift to install, for example {product-version}. The default is the latest version.
 
+|`Configure the use of IMDSv2 for ec2 instances optional/required (optional)`
+|Specify whether all EC2 instances will use both v1 and v2 endpoints of EC2 Instance Metadata Service (IMDS)(optional) or to enforce cluster nodes using IMDSv2 only (required).
```
Comment (bscott-rh) [inline modules/rosa-sts-interactive-cluster-creation-mode-options.adoc:29]:
```suggestion
|Specify whether all EC2 instances will use both v1 and v2 endpoints of EC2 Instance Metadata Service (IMDS)(optional) or only ISDSv2 (required).
```

**Review by bscott-rh — APPROVED** [review]
This looks good! I left a handful of rewording suggestions for clarity and minimalism.

File: modules/rosa-create-objects.adoc [hunk 8]
```diff
@@ -134,6 +134,9 @@ a|--cluster-name <cluster_name>
 |--dry-run
 |Simulates creating the cluster.
 
+|--ec2-metadata-http-tokens string
+|Configures the use of IMDSv2 for EC2 instances, 'optional' or 'required'.
```
[the remark this answers is not in the document]
Comment (mletalie) [reply modules/rosa-create-objects.adoc:138]: Done.

File: modules/rosa-imds-cli.adoc [hunk 9]
```diff
@@ -0,0 +1,37 @@
+// Module included in the following assemblies:
+//
+// * rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-with-customizations.adoc
+
+:_content-type: PROCEDURE
+[id="rosa-imds-cli_{context}"]
+= Enabling Instance Metadata Service in CLI
+
+You can select your Instance Metadata Service (IMDS) type when creating your cluster in your ROSA CLI. You can select to use both IMDSv1 and IMDSv2, or you can specify to only use IMDSv2.
```
[the remark this answers is not in the document]
Comment (mletalie) [reply modules/rosa-imds-cli.adoc:9]: Done.

File: modules/rosa-imds-cli.adoc [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+.Prerequisites
+
+* You installed and configured the latest AWS (`aws`), ROSA (`rosa`), and OpenShift (`oc`) CLIs on your workstation.
+* You logged in to your Red Hat account by using the `rosa` CLI.
```
[the remark this answers is not in the document]
Comment (mletalie) [reply modules/rosa-imds-cli.adoc:14]: Done

File: modules/rosa-imds-cli.adoc [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+.Procedure
+
+. In your terminal, create a ROSA cluster with your specifications by running the following command:
++
+[source,terminal]
+----
+$ rosa create cluster --cluster <name_of_cluster> --ec2-metadata-http-tokens <required_or_optional> <1>
+----
++
+<1> You can provide a value for the `--ec2-metadata-http-tokens` flag, either `required` to enable only IMDSv2 type, or `optional` for a combination of IMDSv1 and IMDSv2. If you do not include this flag, you must select your IMDS type during the cluster creation prompts.
```
[the remark this answers is not in the document]
Comment (mletalie) [reply modules/rosa-imds-cli.adoc:26]: Done.

File: modules/rosa-imds-cli.adoc [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+. Confirm the selection:
++
+[source,terminal]
+----
+? Configure the use of IMDSv2 for ec2 instances optional/required: required
+----
+
+.Verification
+
+* After your cluster has been created, navigate to your cluster *Overview* tab in {cluster-manager-url} to see the *Instance Metadata Service (IMDS)* field that notes your IMDS version support.
```
[the remark this answers is not in the document]
Comment (mletalie) [reply modules/rosa-imds-cli.adoc:37]: Done.

File: modules/rosa-imds-ui.adoc (hunk 2 above)
Comment (mletalie) [reply modules/rosa-imds-ui.adoc:9]: Done

File: modules/rosa-imds-ui.adoc (hunk 3 above)
Comment (mletalie) [reply modules/rosa-imds-ui.adoc:22]: Done.

File: modules/rosa-imds-ui.adoc (hunk 4 above)
Comment (mletalie) [reply modules/rosa-imds-ui.adoc:27]: Done.

File: modules/rosa-imds.adoc (hunk 5 above)
Comment (mletalie) [reply modules/rosa-imds.adoc:16]: Done.

File: modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc (hunk 6 above)
Comment (mletalie) [reply modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc:259]: I like it; makes more sense. Thanks!

File: modules/rosa-sts-interactive-cluster-creation-mode-options.adoc (hunk 7 above)
Comment (mletalie) [reply modules/rosa-sts-interactive-cluster-creation-mode-options.adoc:29]: Done.

File: modules/rosa-create-objects.adoc [hunk 13]
```diff
@@ -134,6 +134,9 @@ a|--cluster-name <cluster_name>
 |--dry-run
 |Simulates creating the cluster.
 
+|--ec2-metadata-http-tokens string
+|Configures the use of IMDSv2 for EC2 instances. Valid values are 'optional' (default) or 'required'.
```
Comment (abhatt-rh) [inline modules/rosa-create-objects.adoc:138]:
```suggestion
|Configures the use of IMDSv2 for EC2 instances. Valid values are `optional` (default) or `required`.
```
Based on OpenShift Docs [markup reference](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference) 

> Use backticks for all non-GUI "system items", including:
> 
> Inline commands, operations, literal values, variables, parameters, settings, flags, environment variables, user input

File: modules/rosa-imds-ui.adoc [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+* You installed and configured the latest AWS (`aws`), ROSA (`rosa`), and OpenShift (`oc`) CLIs on your workstation.
+* You logged in to your Red Hat account by using the `rosa` CLI.
+* You have the permissions to create and manage clusters.
+* You have access to {cluster-manager-url}.
+
+.Procedure
+
+. Log in to the web console.
+. Create a ROSA cluster using your preferences.
+. In the **Create a ROSA Cluster** wizard on the **Cluster settings** -> **Machine pool** page, under the *Instance Metadata Service (IMDS)* section, select whether your machine pools use both IMDSv1 and IMDSv2, or only IMDSv2.
```
Comment (abhatt-rh) [inline modules/rosa-imds-ui.adoc:22]:
```suggestion
. In the *Create a ROSA Cluster* wizard on the *Cluster settings* -> *Machine pool* page, under the *Instance Metadata Service (IMDS)* section, select whether your machine pools use both IMDSv1 and IMDSv2, or only IMDSv2.
```
Based on OpenShift Docs [markup reference](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference)

> Use single asterisks for web console / GUI items (menus, buttons, page titles, etc.).

File: modules/rosa-sts-creating-a-cluster-with-customizations-ocm.adoc [hunk 15]
```diff
@@ -264,6 +264,13 @@ Alternatively, you can set your autoscaling preferences for the default machine
 ** If you deployed your cluster using a single availability zone, select a *Compute node count* from the drop-down menu. This defines the number of compute nodes to provision to the machine pool for the zone.
 ** If you deployed your cluster using multiple availability zones, select a *Compute node count (per zone)* from the drop-down menu. This defines the number of compute nodes to provision to the machine pool per zone.
 
+. Optional: Select an EC2 Instance Metadata Service (IMDS) configuration - optional (default) or required - to enforce use of IMDSv2. For more information regarding IMDS, see link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html[Instance metadata and user data] in the AWS documentation.
```
Comment (abhatt-rh) [inline modules/rosa-sts-creating-a-cluster-with-customizations-ocm.adoc:267]:
```suggestion
. Optional: Select an EC2 Instance Metadata Service (IMDS) configuration - `optional` (default) or `required` - to enforce use of IMDSv2. For more information regarding IMDS, see link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html[Instance metadata and user data] in the AWS documentation.
```

File: modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc [hunk 16]
```diff
@@ -223,46 +223,54 @@ $ rosa create cluster --interactive --sts
...
 ...
 ----
-<1> When creating the cluster, the listed `OpenShift version` options include the major, minor, and patch versions, for example `4.8.9`.
-<2> If you have more than one set of account roles in your AWS account for your cluster version, an interactive list of options is provided.
-<3> Optional: By default, the cluster-specific Operator role names are prefixed with the cluster name and random 4-digit hash. You can optionally specify a custom prefix to replace `<cluster_name>-<hash>` in the role names. The prefix is applied when you create the cluster-specific Operator IAM roles. For information about the prefix, see _Defining an Operator IAM role prefix_.
+<1> When creating the cluster, the listed `OpenShift version` options include the major, minor, and patch versions, for example `4.13.4`.
+<2> Optional: Specify "optional" to configure all EC2 instances to use both v1 and v2 endpoints of EC2 Instance Metadata Service (IMDS). This is the default value. Specify "required" to configure all EC2 instances to use IMDSv2 only.
```
Comment (abhatt-rh) [inline modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc:259]:
```suggestion
<2> Optional: Specify `optional` to configure all EC2 instances to use both v1 and v2 endpoints of EC2 Instance Metadata Service (IMDS). This is the default value. Specify "required" to configure all EC2 instances to use IMDSv2 only.
```

File: modules/rosa-sts-interactive-cluster-creation-mode-options.adoc [hunk 17]
```diff
@@ -25,6 +25,9 @@ The following table describes the interactive cluster creation mode options:
 |`OpenShift version`
 |Select the version of OpenShift to install, for example {product-version}. The default is the latest version.
 
+|`Configure the use of IMDSv2 for ec2 instances optional/required (optional)`
```
Comment (abhatt-rh) [inline modules/rosa-sts-interactive-cluster-creation-mode-options.adoc:28]: Sorry, the use of the value `optional` vs it's use as a choice was a bit confusing. If setting this is optional, then I suggest rewording this for clarity
```suggestion
|Optional:`Configure the use of IMDSv2 for ec2 instances optional/required`
```

**Review by abhatt-rh — COMMENTED** [review]
Hi @mletalie I noticed a few nits during my merge review. Can you PTAL? I can then proceed with the merge.

File: modules/rosa-sts-interactive-cluster-creation-mode-options.adoc (hunk 17 above)
Comment (mletalie) [reply modules/rosa-sts-interactive-cluster-creation-mode-options.adoc:28]: Hello @abhatt-rh,
   Thanks for the suggestion. I am going to keep this one as-is to match the others in the list. I understand your take though, thx!

File: modules/rosa-create-objects.adoc (hunk 13 above)
Comment (mletalie) [reply modules/rosa-create-objects.adoc:138]: Done.

File: modules/rosa-imds-ui.adoc (hunk 14 above)
Comment (mletalie) [reply modules/rosa-imds-ui.adoc:22]: Done

File: modules/rosa-sts-creating-a-cluster-with-customizations-ocm.adoc (hunk 15 above)
Comment (mletalie) [reply modules/rosa-sts-creating-a-cluster-with-customizations-ocm.adoc]: Done

File: modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc (hunk 16 above)
Comment (mletalie) [reply modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc]: Done
